runelite-client/src/main/java/net/runelite/client/rs/ClientLoader.java

```diff
@@ -203,14 +205,24 @@ public Applet load()
 		catch (IOException | ClassNotFoundException | InstantiationException | IllegalAccessException
 			| CompressorException | InvalidHeaderException e)
 		{
+			String message = "Error loading the Old School RuneScape client";
 			if (e instanceof ClassNotFoundException)
 			{
 				log.error("Unable to load client - class not found. This means you"
 					+ " are not running RuneLite with Maven as the client patch"
 					+ " is not in your classpath.");
 			}
+			else if (e instanceof SocketTimeoutException)
+			{
+				log.error("Unable to load client - connection timed out. This is most"
+					+ " likely caused by a Jagex server issue.");
+				message = "Error connecting to the Old School RuneScape servers.";
```

Comment:
This error message is not useful. Maybe "Timeout when connecting to Old School RuneScape servers". Also I think there are probably more errors than just socket timeout?

Reply:
All other errors will default to `Error loading the Old School RuneScape client`. I'm not sure what error messages to use in the other cases. IOException is caused by multiple things but the others aren't

Line 172: `Patch.patch(file.getValue(), bytes, patchOs);` throws the `InvalidHeaderException` & `CompressorException`
Line 201: `Applet rs = (Applet) clientClass.newInstance()` throws the `InstantiationException` & `IllegalAccessException`

I could make generic error messages like `Failed to apply patch` and `Failed to initialize new client instance`?